docs/src/app/components/pages/page-with-nav.jsx

```diff
@@ -68,30 +73,34 @@ let PageWithNav = React.createClass({
         <div style={this.prepareStyles(styles.content)}>
           {this.props.children}
         </div>
-        <div style={this.prepareStyles(styles.secondaryNav)}>
-          <Menu
-            ref="menuItems"
-            zDepth={0}
-            menuItems={this.props.menuItems}
-            selectedIndex={this._getSelectedIndex()}
-            onItemTap={this._onMenuItemClick} />
-        </div>
+        <Menu
+          autoWidth={false}
+          onItemTouchTap={this._handleItemTouchTap}
+          openDirection="bottom-right"
+          style={styles.secondaryNav}
+          valueLink={{
+            value: this.state.selectedIndex,
+            requestChange: this._handleUpdateSelectedIndex,
+          }}
+          width={styles.secondaryNav.width}
+          zDepth={0}>
+          {this.props.menuItems.map((item, index) => {
+            return (<MenuItem
+              key={index}
+              onTouchTap={() => { this.history.pushState(null, item.route); }}
+              primaryText={item.text}
+              value={item.route}
+            />);
+          })}
+        </Menu>
       </div>
     );
   },
 
-  _getSelectedIndex() {
-    let menuItems = this.props.menuItems;
-    let currentItem;
-
-    for (let i = menuItems.length - 1; i >= 0; i--) {
-      currentItem = menuItems[i];
-      if (currentItem.route && this.history.isActive(currentItem.route)) return i;
-    }
-  },
-
-  _onMenuItemClick(e, index, item) {
-    this.history.pushState(null, item.route);
+  _handleUpdateSelectedIndex(e, index) {
+    this.setState({
+      selectedIndex: index,
```

Comment:
Shouldn't we have a default value for this state property?
When reloading a page, the corresponding menu item is not selected.